be/src/storage/tablet_updates.cpp

```diff
@@ -1890,7 +1949,16 @@ void TabletUpdates::_apply_compaction_commit(const EditVersionInfo& version_info
         _set_error(msg);
         return;
     }
-    index.prepare(version, 0);
+    st = index.prepare(version, 0);
+    if (!st.ok()) {
+        manager->index_cache().release(index_entry);
```

Comment:
Using `DeferOp defer([&]() { manager->index_cache().release(index_entry); });` maybe more safe for releasing index entry.

Reply:
done